crates/next-core/src/page_source.rs

```diff
@@ -656,36 +651,60 @@ impl SsrEntryVc {
         } else {
             this.ty
         };
-        let virtual_asset = match ty {
+        let (internal_asset, inner_assets) = match ty {
             SsrType::AutoApi => unreachable!(),
-            SsrType::Api => VirtualAssetVc::new(
-                this.entry_asset.ident().path().join("server-api.tsx"),
-                next_js_file("entry/server-api.tsx").into(),
-            ),
-            SsrType::EdgeApi => VirtualAssetVc::new(
-                this.entry_asset.ident().path().join("server-edge-api.tsx"),
-                next_js_file("entry/server-edge-api.tsx").into(),
-            ),
-            SsrType::Data => VirtualAssetVc::new(
-                this.entry_asset.ident().path().join("server-data.tsx"),
-                next_js_file("entry/server-data.tsx").into(),
+            SsrType::Api => (
+                next_asset("entry/server-api.tsx"),
+                indexmap! {
+                    "INNER".to_string() => entry_asset_page,
+                },
             ),
-            SsrType::Html => VirtualAssetVc::new(
-                this.entry_asset.ident().path().join("server-renderer.tsx"),
-                next_js_file("entry/server-renderer.tsx").into(),
+            SsrType::EdgeApi => {
+                let entry_asset_edge_chunk_group =
+                    this.context.with_transition("next-edge").process(
+                        this.entry_asset,
+                        Value::new(ReferenceType::Entry(EntryReferenceSubType::PagesApi)),
+                    );
```

Comment:
Same comment for applying this transition here: could we avoid adding it to the context altogether?